src/baldr/graphreader.cc

```diff
@@ -36,40 +42,32 @@ tile_gone_error_t::tile_gone_error_t(std::string prefix, baldr::GraphId edgeid)
 }
 
 GraphReader::tile_extract_t::tile_extract_t(const boost::property_tree::ptree& pt) {
+  // A lambda for loading the contents of a graph tile tar from an index file
+  auto index_loader = [this](const std::string& filename, const char* position, size_t size, const char* file_begin) -> decltype(midgard::tar::contents) {
+    // has to be our specially named index.bin file
+    if(filename != "index.bin")
+        return {};
+    
+    // get the info
+    decltype(midgard::tar::contents) contents;
+    midgard::iterable_t<tile_index_entry> index(position, size);
+    for(const auto& entry : index) {
+      // informs the tar that we parsed the index, even though we write a bogus file name
+      auto inserted = contents.insert(std::to_string(entry.tile_id),
+                        std::make_pair(const_cast<char*>(file_begin + entry.offset), entry.size));
+      // store this info for ourselves so we dont have to parse tile names again below
+      tiles[entry.tile_id] = inserted.first->second;
+    }
+  };    
```

Comment:
i basically reverted the changes in this file and added only this bit and the updated call to the tar constructor below